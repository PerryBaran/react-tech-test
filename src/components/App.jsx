import React, { useState } from "react";
import "../styles/app.css";
import Search from "./Search";
import SearchResults from "./SearchResults";

const App = () => {
  const [searchResults, setSearchResults] = useState([]);

  return (
    <div className="app">
      <img
        src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
        alt="NASA logo"
        className="nasa-logo"
      />
      <Search setSearchResults={setSearchResults}/>
      <SearchResults searchResults={searchResults}/>
    </div>
  );
};

export default App;
