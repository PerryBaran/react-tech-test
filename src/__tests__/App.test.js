import { render, screen } from "@testing-library/react";
import App from "../components/App";

describe("App", () => {
  test("snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  })

  test("nasa logo", () => {
    render(<App />);
    const logo = screen.getByAltText("NASA logo");
    
    expect(logo).toHaveAttribute("src", "https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg");
  });
});