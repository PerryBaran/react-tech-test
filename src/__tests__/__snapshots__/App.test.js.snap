// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <img
      alt="NASA logo"
      class="app__nasa-logo"
      src="https://cdn.cnn.com/cnnnext/dam/assets/200424060716-nasa-worm-logo.jpg"
    />
    <input
      class="search__input"
      type="text"
    />
  </div>
</DocumentFragment>
`;
